Test failure for test_join_random

```
seed = 3124334466, lt = ltype.real
    @pytest.mark.parametrize("seed,lt", [(random.getrandbits(32), ltype.bool),
                                         (random.getrandbits(32), ltype.int),
                                         (random.getrandbits(32), ltype.real),
                                         (random.getrandbits(32), ltype.str)])
    def test_join_random(seed, lt):
        random.seed(seed)
        ndata = int(random.expovariate(0.0005))
        nkeys = int(random.expovariate(0.01)) + 1
        st = random.choice(lt.stypes)
        if lt == ltype.bool:
            keys = [True, False]
        elif lt == ltype.int:
            nbits = (6 if st == stype.int8 else
                     12 if st == stype.int16 else
                     24)
            keys = list(set(random.getrandbits(nbits) for _ in range(nkeys)))
        elif lt == ltype.real:
            keys = list(set(random.random() for _ in range(nkeys)))
        else:
            l = int(random.expovariate(0.05)) + 1
            keys = list(set(random_string(l) for _ in range(nkeys)))
        nkeys = len(keys)


dkey = dt.Frame(KEY=keys, VAL=range(nkeys), stypes={"KEY": st})
>       dkey.key = "KEY"

E       datatable.ValueError: Cannot set column as a key: the values are not unique
```

--- a/tests/test_join.py
+++ b/tests/test_join.py
@@ -83,7 +83,11 @@ def test_join_random(seed, lt):
                  24)
         keys = list(set(random.getrandbits(nbits) for _ in range(nkeys)))
     elif lt == ltype.real:
-        keys = list(set(random.random() for _ in range(nkeys)))
+        keys = [random.random() for _ in range(nkeys)]
+        if st == stype.float32:
+            keys = list(set(dt.Frame(keys, stype=st).topython()[0]))
+        else:
+            keys = list(set(keys))
     else:
         l = int(random.expovariate(0.05)) + 1
         keys = list(set(random_string(l) for _ in range(nkeys)))

